test_ask.py

```diff
@@ -3,24 +3,131 @@
 
 
 class TestAsk(unittest.TestCase):
+	# get_root_from_file_path
 	def test_get_root_from_file_path(self):
 		file_path = '/folder/folder/file.txt'
 		self.assertEqual('/folder/folder', ask.get_root_from_file_path(file_path))
 
-	def test_route_path_to_func_name(self):
-		route_paths = [
-			'/path',
-			'/path/subpath',
-			'/path/subpath/',
-			'/path-path'
-		]
-
-		func_names = [
-			'_path',
-			'_path_subpath',
-			'_path_subpath_',
-			'_path_path'
-		]
-
-		for index, route_path in enumerate(route_paths):
-			self.assertEqual(func_names[index], ask.route_path_to_func_name(route_path))
+	# route_path_to_func_name
+	def test_route_path_to_func_name_path(self):
+		test_value = '/path'
+		expected = '_path'
+		self.assertEqual(expected, ask.route_path_to_func_name(test_value))
+
+	def test_route_path_to_func_name_path_subpath(self):
+		test_value = '/path/subpath'
+		expected = '_path_subpath'
+		self.assertEqual(expected, ask.route_path_to_func_name(test_value))
+
+	def test_route_path_to_func_name_path_subpath_slash(self):
+		test_value = '/path/subpath/'
+		expected = '_path_subpath_'
+		self.assertEqual(expected, ask.route_path_to_func_name(test_value))
+
+	def test_route_path_to_func_name_path_dash_path(self):
+		test_value = '/path-path'
+		expected = '_path_path'
+		self.assertEqual(expected, ask.route_path_to_func_name(test_value))
+
+	def test_route_path_to_func_name_path_char(self):
+		test_value = '/path/<path>'
+		expected = '_path_path'
```

Comment:
Expected should be: `_path__path_` since *<* and *>*s also gets converted into underscores.

Reply:
Updated the expected results to reflect this.